dubbo-metrics/dubbo-metrics-api/src/test/java/org/apache/dubbo/metrics/filter/observation/ObservationSenderFilterTest.java

```diff
@@ -0,0 +1,65 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one or more
+ * contributor license agreements.  See the NOTICE file distributed with
+ * this work for additional information regarding copyright ownership.
+ * The ASF licenses this file to You under the Apache License, Version 2.0
+ * (the "License"); you may not use this file except in compliance with
+ * the License.  You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.apache.dubbo.metrics.filter.observation;
+
+import io.micrometer.common.KeyValues;
+import io.micrometer.core.tck.MeterRegistryAssert;
+import io.micrometer.tracing.test.simple.SpansAssert;
+import org.apache.dubbo.common.URL;
+import org.apache.dubbo.rpc.Filter;
+import org.apache.dubbo.rpc.RpcContext;
+import org.apache.dubbo.rpc.cluster.filter.ClusterFilter;
+import org.apache.dubbo.rpc.model.ApplicationModel;
+import org.assertj.core.api.BDDAssertions;
+
+class ObservationSenderFilterTest extends AbstractObservationFilterTest {
+
+    @Override
+    public SampleTestRunnerConsumer yourCode() throws Exception {
+        return (buildingBlocks, meterRegistry) -> {
+            setupConfig();
+            setupAttachments();
+
+            filter.invoke(invoker, invocation);
+
+            BDDAssertions.then(RpcContext.getClientAttachment().getAttachment("X-B3-TraceId")).isNotEmpty();
```

Comment:
Unit test is error in here.
I think DubboClientContext and DubboServerContext should be changed as follows: 
![image](https://user-images.githubusercontent.com/56248584/212853055-0c47a3b3-926b-4177-aaac-2c1a63ab0f8d.png)